### ISSUE TITLE
[Bug] `vtkStringArray.newClone` raises error

### ISSUE THREAD

**Issue by annehaley** [Kitware/vtk-js#3036]
### Bug description

Attempting to copy a populated `vtkStringArray` instance with the `newClone` method results in the following error:
```
Uncaught TypeError: Cannot create vtkStringArray object without: size > 0, values
    at extend (index.js:104:11)
    at constructor (macros.js:1077:5)
    at publicAPI.newClone (index.js:58:5)
```

I am receiving this error when I attempt a `shallowCopy` of a `vtkDataSetAttributes` instance which contains a string array.

### Steps to reproduce

```
import vtkStringArray from 'vtk.js/Sources/Common/Core/StringArray';

const arr = vtkStringArray.newInstance({
  name: 'testArray',
  size: 1,
});
arr.setData(['foo'], 1);
console.log(arr);

// raises error
const copy = arr.newClone();
```

### Expected Behavior

Calling `arr.newClone()` should return a `vtkStringArray` instance. 

I found I can acheive this expected behavior if I make a modification to the `newClone` implementation. I would be glad to create a pull request if someone confirms that this error is unexpected behavior and that this modification should be done. I found two possible modifications that eliminate the error, though I'm not sure which would be best.

Current implementation:
```
publicAPI.newClone = () =>
    newInstance({
      name: model.name,
      numberOfComponents: model.numberOfComponents,
    });
```

Possible modification 1:
```
publicAPI.newClone = () =>
    newInstance({
      name: model.name,
      numberOfComponents: model.numberOfComponents,
      size: model.size,
    });
```

Possible modification 2:
```
publicAPI.newClone = () =>
    newInstance({
      name: model.name,
      numberOfComponents: model.numberOfComponents,
      values: model.values,
    });
```

In summary, it appears that the `newClone` implementation just needs to copy either the `size` attribute or the `values` attribute to the new object in order to eliminate the error.

### Environment

- vtk.js version: 29.10.0
- Browsers: Chrome Version 122.0.6261.111 and Firefox 124.0
- OS: Ubuntu 20.04

**Comment by finetjul** [comment]
Thanks for your report.
`vtkStringArray` should behave like `vtkDataArray`. Therefore it should not create an array with the same size. You might simply want to pass `empty: true` to `newInstance`.

**Comment by annehaley** [comment]
Hi @finetjul, thanks. Do you mean passing `empty: true` to the initial creation of `arr` in my example or passing it to the `newInstance` call within `newClone`? When I tried these, the former still has the error but the latter eliminates the error. If you meant the latter, I can create a PR with that change.

**Comment by finetjul** [comment]
I meant 
```
publicAPI.newClone = () =>
    newInstance({
      name: model.name,
      numberOfComponents: model.numberOfComponents,
      empty: true,
    });
```
Thanks for doing the PR !
